File: C#/RPC/Hayaa.RPC.Componet/Util/IntHelper.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Hayaa.RPC.Service.Util
{
    internal class IntHelper
    {
        public static int ByteArrayToInt(byte[] val, Boolean isBig=true)
        {
            if (isBig)
            {
                byte[] arr = new byte[] { val[3], val[2], val[1], val[0] };
                return BitConverter.ToInt32(arr, 0);

            }
            else
            {
                return BitConverter.ToInt32(val, 0);
            }
        }
        public static byte[] IntToByteArray(int val, Boolean isBig = true)
        {
            if (isBig)
            {
                var arr = BitConverter.GetBytes(val);
                return new byte[] { arr[3], arr[2], arr[1], arr[0] };
            }
            else
            {
                return BitConverter.GetBytes(val);
            }

        }
    }
}

File: Java/v1/src/main/java/hayaa/rpc/common/IntHelper.java
package hayaa.rpc.common;

public class IntHelper {
    public static int byteArrayToInt(byte[] s) {
        return   s[3] & 0xFF |
                (s[2] & 0xFF) << 8 |
                (s[1] & 0xFF) << 16 |
                (s[0] & 0xFF) << 24;
    }
    public static byte[] intToByteArray(int s) {
        return new byte[] {
                (byte) ((s >> 24) & 0xFF),
                (byte) ((s >> 16) & 0xFF),
                (byte) ((s >> 8) & 0xFF),
                (byte) (s & 0xFF)
        };
    }
}
